src/hw/maple/maple.c

```diff
@@ -27,8 +27,37 @@ static void maple_register_device(struct maple *mp, const char *device_type,
 }
 
 void maple_joy_add(struct device *d, int joystick_index) {
-  LOG_INFO("NEW SUPER COOL JOYSTICK ADDED LMAO");
+  if(joystick_index >= MAPLE_NUM_PORTS) {
+    return;
+  }
+  struct maple *mp = (struct maple *)d;
+
```

Comment:
To support this, I think the original joystick_index would need to be stored on each maple_device that gets created.

That way, in the keydown event the incoming joystick_index could be mapped to the correct device.

With that said however, I think it's fine to just always map `joystick_index == maple port` for now, e.g.:

```
CHECK(mp->devices[joystick_index][0] == NULL);
maple_register_device(mp, "controller", joystick_index, 0);
```